### ISSUE TITLE
1.3: Support heading for tasktroubleshooting element

### ISSUE THREAD

**Issue by drmacro** [dita-ot/dita-ot#1771]
Proposal 13096

EDIT: clarification added by @robander, May 4 2016:
DITA 1.3 introduced `<tasktroubleshooting>` as a specialization of `<section>` within task topics.

We currently have the ability to add headings to task sections, using the `args.gen.task.lbl` parameter. This is supported for PDF and X/HTML (not sure about other formats). This feature request is a request to add headings for the new task section as well.

**Comment by robander** [comment]
Just checking on some of these issues - the element is supported with fallback processing (same as most task sections), but doesn't have a generated header when `args.gen.task.lbl` is active.

**Comment by robander** [comment]
Checking again - @drmacro am I correct in thinking that a generated header is the best / only appropriate extra "support" needed for this one? If so, we should tweak the title / description to clarify. And maybe find somebody to translate "Troubleshooting" into our large list of languages...

**Comment by drmacro** [comment]
I think generating a title is the only support required.

**Comment by shudson310** [comment]
I expect that when args.gen.task.lbl is set to "yes", that rendered troubleshooting topics would get labels for:
- Condition
- Cause
- Remedy

Also a label "Troubleshooting" for steptroubleshooting and tasktroubleshooting.

**Comment by shaneataylor** [comment]
@shudson310 I agree this would be very helpful, but wonder if troubleshooting section labels shouldn't be controlled by a separate parameter, maybe a new `args.gen.tbs.lbl`?

Alternatively, maybe a new token-based `args.gen.labels` parameter would provide more flexibility, so users could specify (e.g.) 
```
args.gen.labels = steps result condition remedy tasktroubleshooting
```

Also (and this might be my own quibble with the spec), when (almost always) the problem condition can be stated succinctly, I put it in `<shortdesc>` rather than `<condition>` to assist users who are scanning / searching troubleshooting topics to resolve a problem. I wonder what others think about:

- applying the Condition label to the short description if no `<condition>` is specified?
- processing `<condition>` as a `<shortdesc>` if no `<shortdesc>` exists?

**Comment by robander** [comment]
I've sort of disliked the `args.gen.task.lbl` parameter ever since I introduced it :grimacing: It's verbose, handles an overly specific condition with a command line option, and IMHO has the wrong default value (but that default was purely for backwards compatibility).

With that said ... I'd rather use it than creating a new (even more specific) `args.gen.tbs.lbl`. This a sort of task anyway, so there's a good argument for putting it in with other task labels. 

My gut reaction was that I preferred the idea of a more general `args.gen.labels` parameter ... but we have so many labels, and extensions add their own labels, so in my mind I'm already seeing some exploding complexity. ("How do I set this to create labels for notes, but not for links, and for some task sections, but stop doing the label for sub-topics in the pdf mini-TOC, and ...)

**Comment by robander** [comment]
> Also (and this might be my own quibble with the spec), when (almost always) the problem condition can be stated succinctly, I put it in <shortdesc> rather than <condition> to assist users who are scanning / searching troubleshooting topics to resolve a problem. I wonder what others think about:

I could go with (part of) either option. I think I'd prefer the first, but that might be purely from thinking about it in terms of processing. That is -- shortdesc does a lot of stuff, and I wouldn't want to tweak all of those to be aware of `<condition>`; but assuming that function and applying the label when rendering seems simple / logical.

I also wonder if that feedback has gone back to the OASIS group responsible for troubleshooting? I know they've gotten some feedback and are considering changes for the future.

**Comment by xephon2** [comment]
I agree with @shaneataylor for the token based parameter. This is clean and flexible. One parameter for each label would be to noisy and removing the label parameter completely would make it harder to implement custom plugins.

**Comment by jelovirt** [comment]
Use same approach as with related links. This would make the code backwards compatible.

**Comment by robander** [comment]
> Use same approach as with related links. This would make the code backwards compatible.

@jelovirt do you mean the external `args.rellinks` parameter, the internal `include.rellinks`, or both? Were you thinking these would cover the various task labels + troubleshooting, or _every_ default label? (I'm still thinking it would get out of hand if one `args.gen.labels` parameter tries to handle every label, unless maybe it had logical groups like the `args.rellinks` parameter -- `task` to generate all task labels, `troubleshooting` to generate all TS labels, `reference` to generate labels for the properties table, etc).

**Comment by raducoravu** [comment]
One of our clients also expressed interest in this for the PDF output:

>   The topic <troubleshooting> is badly transformed.  I did not find the XSL code related to the <condition>, <cause> and <remedy> tag transformation. I wish to modify them to our taste.

**Comment by raducoravu** [comment]
Also the HTML 5 output for DITA elements like "steptroubleshooting" and "tasktroubleshooting" does not create any "div" elements for them in the HTML output making it impossible to use CSS to style the content.

**Comment by raducoravu** [comment]
👍 Another one of our end users wanted support for adding labels using "args.gen.task.lbl" for "steptroubleshooting"

**Comment by raducoravu** [comment]
👍 One more end user who needs at least some kind of wrapper element around the HTML content generated from the steptrobleshooting DITA element.
